service/src/main/java/bio/terra/pipelines/stairway/imputation/steps/gcp/FetchOutputsFromDataTableStep.java

```diff
@@ -0,0 +1,95 @@
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This step calls Rawls to fetch outputs from a data table row for a given job and stores them in
+ * the flight's working map. These outputs are considered raw in that they are cloud paths and not
+ * signed urls.
+ */
+public class FetchOutputsFromDataTableStep implements Step {
+
+  private final RawlsService rawlsService;
+  private final SamService samService;
+
+  public FetchOutputsFromDataTableStep(RawlsService rawlsService, SamService samService) {
+    this.rawlsService = rawlsService;
+    this.samService = samService;
+  }
+
+  @Override
+  @SuppressWarnings(
+      "java:S2259") // suppress warning for possible NPE when calling pipelineName.getValue(),
+  //  since we do validate that pipelineName is not null in `validateRequiredEntries`
+  public StepResult doStep(FlightContext flightContext) {
+    String jobId = flightContext.getFlightId();
+
+    // validate and extract parameters from input map
+    var inputParameters = flightContext.getInputParameters();
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        JobMapKeys.PIPELINE_NAME.getKeyName(),
+        RunImputationJobFlightMapKeys.CONTROL_WORKSPACE_BILLING_PROJECT,
+        RunImputationJobFlightMapKeys.CONTROL_WORKSPACE_NAME,
+        RunImputationJobFlightMapKeys.PIPELINE_OUTPUT_DEFINITIONS);
+
+    String controlWorkspaceBillingProject =
+        inputParameters.get(
+            RunImputationJobFlightMapKeys.CONTROL_WORKSPACE_BILLING_PROJECT, String.class);
+    String controlWorkspaceName =
+        inputParameters.get(RunImputationJobFlightMapKeys.CONTROL_WORKSPACE_NAME, String.class);
+    PipelinesEnum pipelineName =
+        inputParameters.get(JobMapKeys.PIPELINE_NAME.getKeyName(), PipelinesEnum.class);
+    List<PipelineOutputDefinition> outputDefinitions =
+        inputParameters.get(
+            RunImputationJobFlightMapKeys.PIPELINE_OUTPUT_DEFINITIONS, new TypeReference<>() {});
+
+    Entity entity;
+    try {
+      entity =
+          rawlsService.getDataTableEntity(
+              samService.getTeaspoonsServiceAccountToken(),
+              controlWorkspaceBillingProject,
+              controlWorkspaceName,
+              pipelineName.getValue(),
+              jobId);
+    } catch (RawlsServiceApiException e) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+
+    Map<String, String> outputs = new HashMap<>();
+    for (PipelineOutputDefinition outputDefinition : outputDefinitions) {
+      String keyName = outputDefinition.getName();
+      String wdlVariableName = outputDefinition.getWdlVariableName();
+      outputs.put(keyName, entity.getAttributes().get(wdlVariableName).toString());
```

Comment:
as we chatted on slack, it _may_ be worth validating the outputs exist and are in fact not empty